### PR TITLE
Fixes #308 Add option to show `Fork me on Github` ribbon

### LINKED ISSUE #308: Add option to display a `Fork me on github` ribbon with link to the repository
### This is a
- [ ] Bug report
- [x] Feature request

### Expected behaviour

There should be a config var which if enabled should produce a `Fork me on github` ribbon in the generated docs.

### PR DESCRIPTION
**Pull request by pri22296**
## Description

Added option to add a `Fork me on github` ribbon.

#### Usage

```yaml
build:
  github_ribbon:
    enable: true
    position: right
    color: red
```

## Related Issue

Fixes #308

## How Has This Been Tested?

Demo deployment: https://yaydocx.herokuapp.com/
Generated docs: https://yaydocx.herokuapp.com/preview/priyamsingh.22296@gmail.com/757d86e0-2fc4-436d-b127-8e517da86642_preview/

## Screenshots (if appropriate):
screenshot 190

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
Form?

**Comment by pri22296** [comment]
Forgot to add WIP label

**Comment by pri22296** [comment]
Two issues here though

- `position: left` doesn't work with `sphinx_rtd_theme`
- With bootstrap based themes(sphinx_fossasia_theme, sphinx_bootstrap_theme), the ribbon isn't placed on the extreme left or right.

screenshot 189

**Comment by pri22296** [comment]
@imujjwal96 @sch00lb0y Please review. Also review the design of `.yaydoc.yml` and the default values whether they should be changed.

**Comment by imujjwal96** [comment]
@pri22296 I don't think we can do anything with the themes. From what it looks, the ribbon is shown over the documentation body!

**Comment by imujjwal96** [comment]
@pri22296 why do we need the enable key? having the base option of `github_ribbon` should be enuff

**Comment by pri22296** [comment]
@imujjwal96 The enable key is needed because other attributes position and color have defaults. So currently there is no way to differentiate whether those values are coming from the yaml file or from the default section. I guess, it's actually an issue with the current implementation of config.py.

**Comment by imujjwal96** [comment]
> . I guess, it's actually an issue with the current implementation of config.py.

Will it require a fix?

File: generate.sh [hunk 1]
```diff
@@ -103,7 +103,11 @@ mkdir -p _themes
 
 # Setting up documentation sources
 print_log "Setting up documentation sources\n"
-sphinx-quickstart -q -v "$VERSION" -a "$AUTHOR" -p "$PROJECTNAME" -t templates/ -d html_theme=$DOCTHEME -d html_logo=$LOGO -d root_dir=$ROOT_DIR -d autoapi_python=$AUTOAPI_PYTHON -d mock_modules=$MOCK_MODULES >>${LOGFILE} 2>>${LOGFILE}
+sphinx-quickstart -q -v "$VERSION" -a "$AUTHOR" -p "$PROJECTNAME" -t templates/ -d html_theme=$DOCTHEME \
```
Comment (imujjwal96) [inline generate.sh]: :+1:

**Comment by pri22296** [comment]
> Will it require a fix?

Actually I'm rewriting the entire code which handles configuration to make it more robust. The current implementation had elements of `Travis implementation`  left from earlier days. I decided it's better to rewrite the whole thing. It would take some time though. I'll also remove the `enable` keyword when i make a PR for it.

**Pull request merged**